pyproject.toml

```diff
@@ -1,6 +1,8 @@
 [tool.scikit-build]
 cmake.minimum-version = "3.26.4"
 cmake.verbose = true
+cmake.args = ["-DCUDAToolkit_ROOT=/usr/local/cuda-12.3", "--debug-find"]
```

Comment:
Though it looks like we are building with CUDA 12.3 here. Do we need CUDA 12.3? Or could we build with CUDA 12.2?

Reply:
The `12.3` version of `libnvjitlink` has a couple of bugfixes we could use. cc @gmarkall for details.

Reply:
[nvbug 4375843](https://nvbugswb.nvidia.com/NVBugs5/redir.aspx?url=/4375843) is one issue that I know of but there may be others.

Reply:
@brandon-b-miller / @gmarkall Is that the same issue tracked in #3 / #4?

Reply:
Think that is a different bug. My understanding is the memory corruption bug was fixed in 12.2u1 whereas conda-forge packages are on 12.2u2

Reply:
@jakirkham, you are correct. The bug ids are 4080381 and 4121608. 12.2u1 should have a fix.

Reply:
Due to cases [like this](https://github.com/rapidsai/pynvjitlink/pull/33#discussion_r1446404877) we actually do need to build with 12.3 (12.latest in general). The bug fixes may be irrelevant since we have to use 12.3 (or 12.latest) to provide a proper compatibility story with pynvjitlink.